File: 0219-contains-duplicate-ii/hashmap/python/solution.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        index_map = {}
        for idx, num in enumerate(nums):
            if num in index_map and idx - index_map[num] <= k:
                return True
            index_map[num] = idx
        return False

File: 0219-contains-duplicate-ii/hashmap/cpp/solution.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map<int, int> indexMap;
        for (int idx = 0; idx < nums.size(); idx++) {
            int num = nums[idx];
            if (indexMap.find(num) != indexMap.end() && idx - indexMap[num] <= k) {
                return true;
            }
            indexMap[num] = idx;
        }
        return false;
    }
};
